### PR TITLE
feat(STONEINTG-454): copy build labels/annotations to new Snapshots

### PR DESCRIPTION
**Pull request by sonam1412**
Copy build labels/annotations prefixed with build.appstudio from pipeline runs to new Snapshots

### CODE REVIEW

File: gitops/snapshot.go [hunk 1]
```diff
@@ -29,6 +29,9 @@ const (
 	// SnapshotTestScenarioLabel contains the name of the Snapshot test scenario.
 	SnapshotTestScenarioLabel = "test.appstudio.openshift.io/scenario"
 
+	// BuildPipelineRunPrefix contains the build pipeline run related labels and annotations
+	BuildPipelineRunPrefix = "build.appstudio"
```
Comment (MartinBasti) [inline gitops/snapshot.go:33]: Can we have full FQDN here?
```suggestion
	BuildPipelineRunPrefix = "build.appstudio.openshift.io"
```

Comment (MartinBasti) [reply gitops/snapshot.go:33]: It would be also compatible with style of this
`PipelinesAsCodePrefix = "pac.test.appstudio.openshift.io"`

Comment (MartinBasti) [reply gitops/snapshot.go:33]: ah okay, I haven't noticed there are mutliple labels

**Comment by sonam1412** [comment]
Changes have been tested on a local cluster. See [here](https://issues.redhat.com/browse/STONEINTG-454?focusedId=22542506&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-22542506)

File: controllers/pipeline/pipeline_adapter_test.go [hunk 2]
```diff
@@ -18,10 +18,11 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"github.com/redhat-appstudio/integration-service/tekton"
 	"reflect"
 	"time"
 
+	"github.com/redhat-appstudio/integration-service/tekton"
+
```
Comment (Josh-Everett) [inline controllers/pipeline/pipeline_adapter_test.go:25]: is this empty line on purpose?

Comment (hongweiliu17) [reply controllers/pipeline/pipeline_adapter_test.go:25]: I think this is formatted by command `go fmt ./...` automatically.

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Review by Josh-Everett — APPROVED** [review]
lgtm

**Review by dheerajodha — CHANGES_REQUESTED** [review]
We're not copying the build labels and annotations to composite snapshots.
Currently, it only copies the PAC annotations/labels from testedSnapshot to compositeSnapshot.

https://github.com/redhat-appstudio/integration-service/blob/bb7334736f2412cc6517e4662f025d5907bad937/controllers/pipeline/pipeline_adapter.go#L576-L578

**Comment by sonam1412** [comment]
> We're not copying the build labels and annotations to composite snapshots. Currently, it only copies the PAC annotations/labels from testedSnapshot to compositeSnapshot.
> 
> https://github.com/redhat-appstudio/integration-service/blob/bb7334736f2412cc6517e4662f025d5907bad937/controllers/pipeline/pipeline_adapter.go#L576-L578
> 
> But since composite snapshots are not created in response to the build PLRs, do we even need the build labels and annotations attached to them?

As discussed on the standup, this is unrelated to STONEINTG-454. We will discuss this point further to decide if it make sense to have build labels/annotations on the composite snapshot and create another story, if needed.

**Review by dheerajodha — APPROVED** [review]

**Review by dirgim — APPROVED** [review]
+1

**Pull request merged**